sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -429,7 +429,7 @@ case class CatalogColumnStat(
    */
   def toMap(colName: String): Map[String, String] = {
     val map = new scala.collection.mutable.HashMap[String, String]
-    map.put(s"${colName}.${CatalogColumnStat.KEY_VERSION}", "1")
+    map.put(s"${colName}.${CatalogColumnStat.KEY_VERSION}", "2")
```

Comment:
For other types it is clear but for `TimestampType` do you propose support wrong old behavior? Since we lose info while storing, I don't see how can we support old behavior for timestamp column in correct way.

Reply:
The existing stats in the table metadata might be wrong, but I don't think we can fail all the queries and ask users to re-generate table stats, because in most cases users won't change the session timezone and they are fine.

Reply:
Added support of previous version in parsing there: https://github.com/apache/spark/pull/23662/files#diff-d129a831466a957545dfc6f8cbcb4b70R492